prow/test/integration/test/webhook_server_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"testing"
+)
+
+// func cleanUpCluster(prowjob string) error {
+// 	err := exec.Command("kubectl", "--context=kind-kind-prow-integration", "delete", "prowjob", prowjob).Run()
+// 	if err != nil {
+// 		return err
+// 	}
+// 	return nil
+// }
+
+const (
+	invalidProwJobID = "test-job"
+)
+
+func TestWebhookServerValidateCluster(t *testing.T) {
+	invalidProwJobPath := "./testdata/invalid_prowjob.yaml"
+	err := createProwJob(invalidProwJobPath, invalidProwJobID, true)
+	if err != nil {
+		t.Errorf("could not properly validate prowjob %s %v", invalidProwJobPath, err)
+	}
+	// err = cleanUpCluster(validProwJobID)
+	// if err != nil {
+	// 	t.Errorf("could not delete prowjob %s %v", validProwJobID, err)
+	// }
+}
+
+func createProwJob(path string, prowjobID string, shouldFail bool) error {
```

Comment:
It does not seem like this should be a separate function. This is doing a lot of the test logic. Just combine this with TestWebhookServerValidateCluster

Reply:
gonna do the same for the valid prowjob as well, there will be a lot of duplicate code.

Reply:
actually ill move it in instead